### PR TITLE
add debian jenkins node to Vagrant

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
@ekohl I wasn't sure what the best way was to install Puppet here, first I tried if/else in the original provisioner, then I overwrote it in the debian box. Neither feels too clean, but here we are :)

File: Vagrantfile [hunk 1]
```diff
@@ -50,6 +50,22 @@ Vagrant.configure("2") do |config|
...
+    override.vm.hostname = "jenkins-node-debian10"
+    override.vm.box = "debian/buster64"
+
+    override.vm.provider "libvirt" do |libvirt|
+      libvirt.memory = "4096"
+    end
+    config.vm.provision "install puppet", type: "shell", inline: <<-SHELL
+      . /etc/os-release
+      wget https://apt.puppet.com/puppet6-release-${VERSION_CODENAME}.deb
+      apt-get install -y ./puppet6-release-${VERSION_CODENAME}.deb
```
Comment (ekohl) [inline Vagrantfile:63]: TIL: apt-get can install local files

File: Vagrantfile [hunk 2]
```diff
@@ -50,6 +50,22 @@ Vagrant.configure("2") do |config|
...
 
+  config.vm.define "jenkins-node-debian10" do |override|
+    override.vm.hostname = "jenkins-node-debian10"
+    override.vm.box = "debian/buster64"
+
+    override.vm.provider "libvirt" do |libvirt|
+      libvirt.memory = "4096"
+    end
+    config.vm.provision "install puppet", type: "shell", inline: <<-SHELL
+      . /etc/os-release
```
Comment (ekohl) [inline Vagrantfile:61]: I like `/etc/os-release` :)

**Review by ekohl — APPROVED** [review]

File: Vagrantfile (hunk 1 above)
Comment (evgeni) [reply Vagrantfile:63]: Yeah, and contrary to `dpkg` it will resolve dependencies when doing so ;)

**Pull request merged**